src/global_checks.jl

```diff
@@ -0,0 +1,56 @@
+"""
+    _parameters(type)
+Extracts the type-parameters of the `type`.
+e.g. `_parameters(Foo{A, B, C}) == [A, B, C]`
+"""
+_parameters(sig::UnionAll) = _parameters(sig.body)
+_parameters(sig::DataType) = sig.parameters
+_parameters(sig::Union) = Base.uniontypes(sig)
+
+"""
+    test_method_tables_sensibility()
```

Comment:
is "sensibility" really the word, rather than just "sense"?

Also maybe just
```suggestion
    test_method_tables()
```
or even have separate methods
```suggestions
    test_method_table(frule|rrule)
```